pkg/lib/web/user_agent.go

```diff
@@ -1,33 +1,53 @@
 package web
 
 import (
+	"strconv"
+
 	"github.com/authgear/authgear-server/pkg/api/model"
 )
 
 // The name is borrowed from https://github.com/browserslist/browserslist
 type RecognizedMobileDeviceType string
 
+type RecognizedMobileDevice struct {
+	Type                 RecognizedMobileDeviceType `json:"type"`
+	OSVersionMajorString string                     `json:"os_version_major_string"`
+	OSVersionMinorString string                     `json:"os_version_minor_string"`
+	OSVersionPatchString string                     `json:"os_version_patch_string"`
+	OSVersionMajorInt    int                        `json:"os_version_major_int"`
+	OSVersionMinorInt    int                        `json:"os_version_minor_int"`
+	OSVersionPatchInt    int                        `json:"os_version_patch_int"`
```

Comment:
Remove the JSON tags. We do not marshal this struct as JSON.

Reply:
ok